forch/faucet_state_collector.py

```diff
@@ -780,14 +780,15 @@ def _get_egress_state_detail(self, links):
         links_status = set()
         egress_name = None
         for key, status in links.items():
-            links_status.add(LacpState.LacpState.Value(status.get(LINK_STATE)))
-            if status.get(LINK_STATE) == LacpState.LacpState.Name(LacpState.active):
+            link_state = status.get(LINK_STATE)
+            links_status.add(LacpState.LacpState.Value(link_state))
+            if link_state == LacpState.LacpState.Name(LacpState.active):
                 egress_name = key
-            if status.get(LINK_STATE) == LacpState.LacpState.Name(LacpState.down):
+            elif link_state == LacpState.LacpState.Name(LacpState.down):
                 link_down = key
-        if links_status == set([LacpState.active, LacpState.up]):
+        if links_status == set([LacpState.active, LacpState.up, LacpState.unknown]):
```

Comment:
Links status set is basically the status of egress links, right? Thy would we have a third link?

Reply:
It's doing a set comparison. Basically it collects a list of all the "allowed" state, and as long as they match the set it goes.  So, it's basically saying that ALL links need to be one of { active, up, unknown } -- notice that down or broken or whatever isn't in there.